src/components/Bridge/Bridge.tsx

```diff
@@ -0,0 +1,181 @@
+import { memo, useCallback, useMemo, useState } from 'react';
+import { Button, Icon, Typography, ValueLabel } from '../shared';
+import PoweredBy from './PoweredBy';
+import NetworkSelector from './NetworkSelector';
+
+import './Bridge.scss';
+import { SUPPORTED_BRIDGE_NETWORKS, USD_UI_PRECISION } from '../../constants';
+import { SupportedBridgeNetwork } from '../../interfaces';
+import { ButtonWrapper, TokenLogo } from 'tempus-ui';
+import AmountInput from './AmountInput';
+import { R_TOKEN } from '@raft-fi/sdk';
+import { Decimal } from '@tempusfinance/decimal';
+import { formatCurrency } from '../../utils';
+import { useConnectWallet } from '@web3-onboard/react';
+
+const Bridge = () => {
+  const [, connect] = useConnectWallet();
+  const [fromNetwork, setFromNetwork] = useState<SupportedBridgeNetwork>('ethereum');
+  const [toNetwork, setToNetwork] = useState<SupportedBridgeNetwork>('base');
+  const [amount, setAmount] = useState<string>('');
+
+  const amountDeciaml = useMemo(() => Decimal.parse(amount, 0), [amount]);
+
+  // TODO: hardcode balance for now
+  const fromBalance = useMemo(() => new Decimal(123), []);
+  const toBalance = useMemo(() => new Decimal(123), []);
+  const toBalanceAfter = useMemo(() => toBalance.add(amountDeciaml), [amountDeciaml, toBalance]);
+
+  const fromBalanceFormatted = useMemo(
+    () =>
+      formatCurrency(fromBalance, {
+        currency: R_TOKEN,
+        fractionDigits: USD_UI_PRECISION,
+      }),
+    [fromBalance],
+  );
+  const toBalanceFormatted = useMemo(
+    () =>
+      formatCurrency(toBalance, {
+        currency: R_TOKEN,
+        fractionDigits: USD_UI_PRECISION,
+      }),
+    [toBalance],
+  );
+  const toBalanceAfterFormatted = useMemo(
+    () =>
+      formatCurrency(toBalanceAfter, {
+        currency: R_TOKEN,
+        fractionDigits: USD_UI_PRECISION,
+      }),
+    [toBalanceAfter],
+  );
+
+  const onSwapNetwork = useCallback(() => {
+    setFromNetwork(toNetwork);
```

Comment:
There is a chance we have a race condition here.

I would store toNetwork and fromNetwork in new variables, and then update state using these variables.
```
const fromNetworkCurrent = fromNetwork;
const toNetworkCurrent = toNetwork;

setFromNetwork(toNetworkCurrent);
setToNetwork(fromNetworkCurrent);
```

Reply:
as discussed, race condition wont happen in this case. leave it as it is.